server/pages/index.js

```diff
@@ -128,6 +128,13 @@ function getDefaultContext( request ) {
 		initialServerState = getInitialServerState( serializeCachedServerState );
 	}
 
+	const protocol = request.get( 'X-Forwarded-Proto' ) === 'https' ? 'https' : 'http';
```

Comment:
Can't we use `request.protocol` here? (Disclaimer -- I just did a quick search that turned up [this SO answer](http://stackoverflow.com/a/10185427).)